### PR TITLE
Refactor static fire data to enable future use of GDP data in SPITFIRE

### PR DESCRIPTION
**Pull request by glemieux**
### Description of changes

This PR implements the refactor detailed by @ekluzek in issue #1104 to enable anthropogenic fire suppression as described by @jkshuman in https://github.com/NGEET/fates/issues/673.

### Specific notes

The refactor moves the GDP, peat, and ag-fires variables and surface dataset read subroutine from CNVegStateType.F90 to FireDataBaseType.F90 into `fire_base_type` so that this data will be available to the extended `fates_fire_base_type`.

Contributors other than yourself, if any: @ekluzek, @jkshuman 

CTSM Issues Fixed (include github issue #): #1104 

Are answers expected to change (and if so in what way)? No

Any User Interface Changes (namelist or namelist defaults changes)? A new spitfire mode has been added, although this will likely need future refinement based on planned future spitfire model updates.

Testing performed, if any:
(List what testing you did to show your changes worked as expected)
(This can be manual testing or running of the different test suites)
(Documentation on system testing is here: https://github.com/ESCOMP/ctsm/wiki/System-Testing-Guide)
(aux_clm on cheyenne for intel/gnu and izumi for intel/gnu/nag/pgi is the standard for tags on master)

**NOTE: Be sure to check your coding style against the standard
(https://github.com/ESCOMP/ctsm/wiki/CTSM-coding-guidelines) and review
the list of common problems to watch out for
(https://github.com/ESCOMP/CTSM/wiki/List-of-common-problems).**

### CODE REVIEW

**Comment by glemieux** [comment]
Note that the new anthro suppression spitfire switch is implemented in a very simple manner so as to test passing the relevant gdp data to fates.  Any future update to the spitfire mode selection will likely need refinement since the use of anthropogenic suppression models will need to coordinate with different ignition source modes.

**Comment by glemieux** [comment]
The hand off of the gdp data from ctsm to fates was tested using https://github.com/glemieux/fates/tree/fire-gdp-simpletest.  Results of the test can be found here: https://github.com/glemieux/fates-jupyter/blob/main/gdp-refactor/gdp-check.ipynb

**Comment by glemieux** [comment]
All test pass b4b against `ctsm5.1.dev061` with the exception of the gnu compiler fates testmods failing to RUN very early.  These were run with fates tag `sci.1.48.0_api.17.0.0`.  Looking at the cesm.logs, it appears to be failing in a manner very similar to https://github.com/ESCOMP/CTSM/issues/1545.  Based on that experience, I reran the gnu compiler fates tests with `sci.1.49.1_api.17.0.0`, which includes the fix from https://github.com/NGEET/fates/pull/807.  Doing so resulted in the tests passing.  This is odd to me as these tests passed in the original ctsm5.1.dev061 baseline, which presumably was run with `sci.1.48.0_api.17.0.0`.  So I'm not sure what could have predicated these failures.

Original test folder: `/glade/u/home/glemieux/scratch/ctsm-tests/tests_pr1536_ctsm-gdprefactor-aux_clm`

**Comment by glemieux** [comment]
All expected tests pass b4b against `ctsm5.1.dev064`:
- cheyenne: `/glade/u/home/glemieux/scratch/ctsm-tests/tests_pr1536-aux_clm`
- izumi: `/scratch/cluster/glemieux/ctsm-tests/tests_pr1536-aux_clm`

File: src/utils/clmfates_interfaceMod.F90 [hunk 1]
```diff
@@ -2426,6 +2446,7 @@ subroutine Init2(this, bounds, NLFilename)
 
     call t_startf('fates_init2')
 
+    write(iulog,*) 'Init2: calling FireInit'
```
Comment (ekluzek) [inline src/utils/clmfates_interfaceMod.F90:2449]: I'm not sure we want this write statement here. But, with testing done, we won't take it out for now. It doesn't hurt much either, although I do suspect it should at least be in a "if ( masterproc)" statement. So we should correct this in an upcoming tag.

**Review by ekluzek — APPROVED** [review]

**Pull request merged**